tests/test_diceware.py

```diff
@@ -317,6 +317,15 @@ def test_main_infile_stdin(self, argv_handler, capsys):
         out, err = capsys.readouterr()
         assert out == 'Word1Word1\n'
 
+    def test_main_infile_nonexisting(self, argv_handler, capsys):
+        # main() prints error if file does not exist
+        sys.argv = ['diceware', 'nonexisting']
+        with pytest.raises(SystemExit) as exc_info:
+            main()
+        assert exc_info.value.code == 1
+        out, err = capsys.readouterr()
+        assert out == "The file 'nonexisting' does not exist.\n"
```

Comment:
You are looking in stdout for stderr output. This will hardly work ;)

Reply:
The message was not in the `stderr` as well :(  It gets captured in caplog, if I have understood it correctly.

Reply:
`out, err` are captured `stdout`, `stderr`. Hence the names.
`assert err == "the file..."` (instead of `out==...`) should therefore do the trick.